account_mtd/models/mtd_hello_world.py

```diff
@@ -0,0 +1,406 @@
+        )
+        headers = {
+            'Content-Type': 'application/json',
+            }
+        self._logger.info(
+            "(Step 2) exchange authorisation code - headers which will be "
+            "sent in the request:-  {}".format(headers)
+        )
+        req = requests.post(token_location_uri, data=json.dumps(data_user_info), headers=headers)
+        response_token = json.loads(req.text)
+        self._logger.info(
+            "(Step 2) exchange authorisation code - received " 
+            "response of the request:- {}, and its text:- {}".format(req, response_token)
+        )
+        if req.ok:
+            # get the record which we created when sending the request and update the response_received column
+            # As this determines whether we can place another request or not
+            record_tracker = self.env['mtd.api_request_tracker'].search([
+                ('request_sent', '=', True),
+                ('response_received', '=', False)
+            ])
+            record_tracker.response_received = True
+            if not api_token:
+                api_token = self.env['mtd.api_tokens'].search([('authorisation_code', '=', auth_code)])
+            self._logger.info(
+                "(Step 2) exchange authorisation code - api_token table id where info is stored:- {}".format(api_token)
+            )
+            api_token.access_token = response_token['access_token']
+            api_token.refresh_token = response_token['refresh_token']
+            api_token.expires_in = json.dumps(response_token['expires_in'])
+            api_token.access_token_recieved_date = datetime.now()
+            version = self._json_command('version', record_id=record_id)
+            return version
+        else:
+            construct_text = (
+                    "Date {}     Time {} \n".format(datetime.now().date(), datetime.now().time()) +
+                    "Sorry. The connection failed ! \n" +
+                    "Please check the log below for details. \n\n" +
+                    "Connection Status Details: \n" +
+                    "Request Sent: \n{} \n\n".format(token_location_uri) +
+                    "Error Code:\n{} \n\n".format(req.status_code) +
+                    "Response Received: \n{}".format(response_token['message'])
+            )
+            self._logger.info(
+                "(Step 2) exchange authorisation code - log:- {}".format(construct_text)
+            )
+            action = self.env.ref('account_mtd.action_mtd_hello_world')
+            menu_id = self.env.ref('account_mtd.submenu_mtd_hello_world')
+            self._logger.info(
+                "(Step 2) exchange authorisation code - redirect URI :- " +
+                "/web#id={}&view_type=form&model=mtd.hello_world".format(record_id) +
+                "&menu_id=72&action={}".format(menu_id.id, action.id)
+            )
+            return werkzeug.utils.redirect(
```

Comment:
Is this a button handler?  If so you can do action redirects by returning a dictionary and don't need to do manual URI redirects.  I'll dig out examples if you need them.